articles/115_topic_and_service_name_mapping.md

```diff
@@ -0,0 +1,313 @@
+A topic name may be preceded by a `rostopic://` scheme prefix, and a service name may be preceded by a `rosservice://` scheme prefix.
+For example, the absolute name `/foo` could also be represented as a topic with `rostopic:///foo` or as a service with `rosservice:///foo`.
+Note the triple forward slash (`/`), which is a similar style as the `file://` scheme.
+A relative name `foo/bar` could would be represented (as a topic) with `rostopic://foo/bar`.
+
+### ROS 2 Name Examples
+
+For example, these are valid names:
+
+| `foo`       | `abc123`            | `_foo`               | `Foo` | `BAR`       |
+| `~`         | `foo/bar`           | `~/foo`              | `/~`  | `foo/~/bar` |
+| `foo/_/bar` | `rosservice:///foo` | `rostopic://foo/bar` |       |             |
+
+But these are not valid names:
+
+| `123abc`   | `123`  | `__foo` | `foo__bar` | `foo bar`  |
+| `foo__`    | ` `    | `foo_`  | `foo//bar` | `foo/`     |
+| `foo_/bar` | `~foo` | `foo~`  | `foo~/bar` | `foo/~bar` |
+
+### Namespaces
+
+Topic and service names:
+
+- may be split into tokens using a forward slash (`/`) as a delimiter
+- must not end with a forward slash (`/`)
+
+The last token is the topic or service name, and any preceding tokens comprise the namespace of the topic or service.
+
+For example, the topic name `/foo/bar/baz` comprises of a topic or service named `baz` in the `/foo/bar` namespace.
+In another example, the name `/foo` does not use splitting into tokens, so it comprises of a topic or service named `foo` in the `/` namespace (the root namespace).
+
+Topic and service names:
+
+- may be specified as absolute or relative
+- must start with a forward slash (`/`) if they are absolute
+
+An absolute name begins with a forward slash (`/`) and does not respect namespaces, i.e. it can be considered "global".
+A relative name does not begin with a forward slash (`/`) and does respect the namespace of the node which created them.
+
+Relative names are appended to the namespace of the node which creates them.
+For example, if the node is put into a namespace `/ping/pong` and the topic or service name is `foo/bar`, then the absolute name will become `/ping/pong/foo/bar`.
+However, if the topic or service name is `/foo/bar`, then the absolute name will stay just `/foo/bar`, ignoring the node's namespace.
+
+### Name Tokens
+
+Name token's are the strings between the namespace delimiters, e.g. the tokens of the topic or service `/foo/bar/baz` are `foo`, `bar`, and `baz`.
+
+Topic and service name tokens:
+
+- must not be empty, e.g. the name `//bar` is not allowed
+
+  - rational: it removes the chance for accidental `//` from concatenation; also if the implementation chooses to use a trailing underscore (`_`) then it prevents `foo_/_bar` and `foo//bar` from being ambiguous
```

Comment:
Yeah, I was thinking about how Connext does Services, where if you give it `foo` it will create to topics called `foo_Request` and `foo_Response` (or something like that). So if we give it `foo_s_` (as was proposed in an alternative) it would do `foo_s__Request`. But either way this is a weak part of the rational for sure. I'm agnostic to removing it, maybe I could get another +1 for removing it from someone else?